src/main/java/org/ibp/api/java/impl/middleware/germplasm/GermplasmServiceImpl.java

```diff
@@ -170,6 +180,60 @@ private void addParentsFromPedigreeTable(final Map<Integer, GermplasmSearchRespo
 		}
 	}
 
+	private void addHasProgenyAttribute(final Map<Integer, GermplasmSearchResponse> responseMap,
+		final GermplasmSearchRequest germplasmSearchRequest) {
+		final List<String> addedColumnsPropertyIds = germplasmSearchRequest.getAddedColumnsPropertyIds();
+		if (addedColumnsPropertyIds == null || addedColumnsPropertyIds.isEmpty()
+			|| !addedColumnsPropertyIds.contains(ColumnLabels.HAS_PROGENY.getName())) {
+			return;
+		}
+
+		final Set<Integer> gidsOfGermplasmWithDescendants =
+			this.germplasmService.getGidsOfGermplasmWithDescendants(new ArrayList<>(responseMap.keySet()));
+
+		for (final Map.Entry<Integer, GermplasmSearchResponse> entry : responseMap.entrySet()) {
```

Comment:
This whole for look can be simplified as: 
`responseMap.forEach((gid, response) -> response.setHasProgeny(gidsOfGermplasmWithDescendants.contains(gid)));`